lib/features/auto-place/AutoPlaceUtil.js

```diff
@@ -109,133 +110,124 @@ export function getConnectedAtPosition(source, position, element) {
 }
 
 /**
-* Compute best distance between source and target,
-* based on existing connections to and from source.
+* Compute optimal distance between source and target based on existing connections to and from source.
 *
 * @param {djs.model.Shape} source
-* @param {string} axis
-* @param {Function} [filter]
 * @param {Object} [hints]
-* @param {djs.model.Shape} [hints.connectionTarget]
-* @param {djs.model.Shape} [hints.maxDistance]
+* @param {number} [hints.defaultDistance]
+* @param {string} [hints.direction]
+* @param {Function} [hints.filter]
+* @param {Function} [hints.getWeight]
+* @param {number} [hints.maxDistance]
+* @param {string} [hints.reference]
```

Comment:
From what I see the `{String}` reference is plain wrong. 

Proposal: Can we make that simple type conversion `{String}` -> `{string}` and friends once for the whole code base as a follow up? That would be amazing.

Could also think about applying it to:

* bpmn-js
* dmn-js
* Camunda Modeler

Reply:
> From what I see the `{String}` reference is plain wrong.

What do you mean?

Reply:
`String` is an object wrapper around the primitive type `string`. Cf. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String#Distinction_between_string_primitives_and_String_objects

Reply:
Same applies to other primitives.

Reply:
> What do you mean?

I mean that it is safe to reference `{string}` in all the places where we referenced `{String}` in the past. Same applies for `Number` and `Boolean`.

I mention this as I see you gradually changing the code base (fixing those docs) rather than doing a one-off fix that corrects this across the whole repository.

If `string` is the type to mention, let's migrate the whole project at once.